### PR TITLE
New model for work titles

### PR DESCRIPTION
**Pull request by kwahlin**
See [LXL-3880](https://jira.kb.se/browse/LXL-3880) for more info

### CODE REVIEW

File: whelk-core/src/main/groovy/whelk/converter/marc/NormalizeWorkTitlesStep.groovy [hunk 1]
```diff
@@ -0,0 +1,65 @@
+package whelk.converter.marc
```
Comment (olovy) [inline whelk-core/src/main/groovy/whelk/converter/marc/NormalizeWorkTitlesStep.groovy:1]: Would like a summary here of what the script does.

**Review by olovy — APPROVED** [review]
I think it was a bit hard to follow the script. The giant if-else structure could be improved.

See comments, below, otherwise :ship:

File: whelktool/scripts/cleanups/2022/06/lxl-3880-shuffle-work-titles.groovy [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+                }
+                if (langsAdded) {
+                    work[LANGUAGE] = worklang
+                    extraLang.println([shortId, langsAdded, worklang].join('\t'))
+                }
+                hub.remove(LANGUAGE)
+            }
+            // Move remaining properties (except @type) to the same place as the title
+            hub.removeAll {
+                it.key == '@type' ?: moveProperty(shortId, it.key, expressionOf, target)
```
[the remark this answers is not in the document]
Comment (kwahlin) [reply whelktool/scripts/cleanups/2022/06/lxl-3880-shuffle-work-titles.groovy]: It fails if any of the properties already exists on the target. These cases have been cleaned up manually but there is a possibility that new records keep coming in with this bad structure.

My thinking was that it doesn't matter if it fails, since it will be reported in 'property-already-exists.tsv' (for manual handling) but yeah, maybe it's better to not save the update at all if it partly fails. See [3d2293a](https://github.com/libris/librisxl/pull/1130/commits/3d2293a30d7cbece0b12adbc2bdf9e99c6f7bd7a)

File: whelktool/scripts/cleanups/2022/06/lxl-3880-shuffle-work-titles.groovy [hunk 3]
```diff
@@ -0,0 +1,400 @@
...
+        }
+
+        // Move expressionOf content primarily to translationOf, otherwise to instanceOf
+        def target = work.find { it.key == TRANSLATION_OF } ?: instanceOf
+
+        Map hub = asList(expressionOf.value)[0]
+        if (hub[ID]) {
+            expressionOf.value = loadThing(hub[ID])
+            if (!expressionOf.value) {
+                brokenLinks.println([shortId, hub[ID]].join('\t'))
```
[the remark this answers is not in the document]
Comment (kwahlin) [reply whelktool/scripts/cleanups/2022/06/lxl-3880-shuffle-work-titles.groovy]: > I think it was a bit hard to follow the script. The giant if-else structure could be improved.
> 
> See comments, below, otherwise ship

Agree that the structure is messy (would be much easier if it wasn't for the detailed reporting). _Tiny_ improvement in [3d2293a](https://github.com/libris/librisxl/pull/1130/commits/3d2293a30d7cbece0b12adbc2bdf9e99c6f7bd7a).

File: whelktool/scripts/cleanups/2022/06/lxl-3880-shuffle-work-titles.groovy [hunk 4]
```diff
@@ -0,0 +1,407 @@
+import whelk.filter.LanguageLinker
```
Comment (olovy) [inline whelktool/scripts/cleanups/2022/06/lxl-3880-shuffle-work-titles.groovy:1]: Would like a summary here of what the script does

Comment (kwahlin) [reply whelktool/scripts/cleanups/2022/06/lxl-3880-shuffle-work-titles.groovy:23]: See [370b3f3](https://github.com/libris/librisxl/pull/1130/commits/370b3f3aec4fa3257e9274be4942514a0d89d813)

Comment (olovy) [reply whelktool/scripts/cleanups/2022/06/lxl-3880-shuffle-work-titles.groovy:23]: :+1:

**Review by niklasl — APPROVED** [review]
Awesome! :star: :ship: 

(Feel free to do the `traverse` with closure-thing we discussed. :slightly_smiling_face:  )
